cpu/cpu.go

```diff
@@ -182,26 +196,43 @@ func (p *Plugin) CollectMetrics(metricTypes []plugin.MetricType) ([]plugin.Metri
 		p.snapMetricsNames, p.procStatMetricsNames); err != nil {
 		return nil, err
 	}
+	ts := time.Now()
+	prefix := filepath.Join(vendor, fs, pluginName)
 	for _, metricType := range metricTypes {
 		ns := metricType.Namespace()
 		if len(ns) != maxNamespaceSize {
 			return nil, fmt.Errorf("Incorrect namespace length (len = %d)", len(ns))
 		}
-
-		val, err := getMapValueByNamespace(p.stats[ns.Strings()[3]], ns.Strings()[4:])
-
-		if err != nil {
-			return metrics, err
-		}
-
-		metric := plugin.MetricType{
-			Namespace_: ns,
-			Data_:      val,
-			Timestamp_: time.Now(),
+		if ns[len(ns)-2].Value == "*" {
+			for cpuId, cpuStats := range p.stats {
+				for k, v := range cpuStats {
+					if ns[len(ns)-1].Value == k && v != nil {
+						ns1 := core.NewNamespace(strings.Split(prefix, string(os.PathSeparator))...).
+							AddStaticElement(cpuId).
```

Comment:
Hello @obourdo, consider do not change incoming dynamic metric to static, just set a value (cpuId) to appropriate element of namespaces as below:
```
rns := make([]core.NamespaceElement, len(ns))
copy(rns, ns)
rns[len(ns)-1].Value = fmt.Sprintf("%d", cpuId)
```
where `rns` (returned namespace) is equivalent of your `ns1`

snap-plugin-collector-mock1 is a good example how it works (see [line 56-58](https://github.com/intelsdi-x/snap/blob/master/plugin/collector/snap-plugin-collector-mock1/mock/mock.go#L56)

This approach (leave metric as dynamic but with spiecified value, e.g. cpuID) causes that snap publisher can recognize this metric as dynamic and move its dynamic element(s) of namespace to tag. @jcooklin, could you confirm that my reasoning is correct? Thanks.